File: 0598_range_addition_ii/max_count.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int maxCount(int m, int n, int[][] ops) {
        if (ops == null || ops.length == 0) {
            return m * n;
        }
        int row = m, col = n;
        for (int i = 0; i < ops.length; i++) {
            row = Math.min(row, ops[i][0]);
            col = Math.min(col, ops[i][1]);
        }
        return row * col;
    }
}

File: 0598_range_addition_ii/max_count.py
from math import prod
from typing import List


def max_count(m: int, n: int, ops: List[List[int]]) -> int:
    """
    Time: O(n)
    Space: O(n)
    """
    return prod([min(z) for z in zip(*ops)]) if ops else m * n